tuning/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,204 @@
+# Copyright The FMS HF Tuning Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Standard
+import json
+import logging
+import os
+
+# Third Party
+from transformers.integrations import MLflowCallback  # pylint: disable=import-error
+
+# Local
+from .tracker import Tracker
+from tuning.config.tracker_configs import MLflowConfig
+
+MLFLOW_RUN_URI_EXPORT_FILENAME = "mlflow_tracker.json"
+
+
+class RunIDExporterMlFlowCallback(MLflowCallback):
+    """
+    Custom MlflowCallBack callback is used to export run id from Mlflow
+    as soon as it is created, which is on setup().
+    """
+
+    run_uri_export_path: str = None
+
+    # Override ml flow callback setup function
+    # Initialise mlflow callback and export Mlflow's run url.
+    # Export location is MlflowConfig.mlflow_run_uri_export_path if it is passed
+    # or, training_args.output_dir/mlflow_tracker.json if output_dir is present
+    # Exported url looks like '{"run_uri":"<uri>"}' in the file.
+    # url is not exported if both paths are invalid
+    def setup(self, args, state, model):
+        """Override the `setup` function in the `MLflowCallback` callback.
+
+            This function performs the following steps:
+            1. Calls `MLFlowCallBack.setup` to
+                initialize internal `mlflow` structures.
+            2. Exports the `Mlflow` run uri:
+                - If `MLflowConfig.mlflow_run_uri_export_path` is provided, the uri
+                  is exported to `mlflow_run_uri_export_path/mlflow_tracker.json`
+                - If `MLflowConfig.mlflow_run_uri_export_path` is not provided but
+                    `args.output_dir` is specified, the uri is exported to
+                    `args.output_dir/mlflow_tracker.json`
+                - If neither path is valid, the uri is not exported.
+
+            The exported uri is formatted as '{"run_uri":"<uri>"}'.
+
+        Args:
+            For the arguments see reference to transformers.TrainingCallback
+        """
+        super().setup(args, state, model)
+
+        active_run = self._ml_flow.active_run()
+        if not active_run:
+            return
+
+        active_run_info = active_run.info
+        if active_run_info:
+            experiment_id = active_run_info.experiment_id
+            experiment_url = f"<host_url>/#/experiments/{experiment_id}"
+            run_id = active_run_info.run_id
```

Comment:
Is it called run_id in MLFlow?

Reply:
Yes - https://github.com/mlflow/mlflow/blob/master/mlflow/entities/run_info.py#L44